tests/integration/cattletest/core/test_container.py

```diff
@@ -40,6 +40,384 @@ def test_container_build(super_client, context, client):
     assert image.data.fields.build.rm
 
 
+def test_container_registry_default(super_client, client, context,
+                                    admin_user_client):
+    id = 'default.registry'
+    default_registry = admin_user_client.by_id_setting(id)
+    id = 'whitelist.registries'
+    whitelist_registries = admin_user_client.by_id_setting(id)
+
+    # case 1: whitelist set to null, all cases pass
+    whitelist_registries = admin_user_client.update(whitelist_registries,
+                                                    value='')
+    uuid = "sim:{}".format(random_num())
+    container = client.create_container(imageUuid=uuid,
+                                        startOnCreate=False)
+
+    container = client.wait_success(container)
+
+    assert_fields(container, {
+        "type": "container",
+        "state": "stopped",
+        "imageUuid": uuid,
+        "firstRunning": None,
+    })
+
+    uuid = "namespace/sim:{}".format(random_num())
+    container = client.create_container(imageUuid=uuid,
+                                        startOnCreate=False)
+
+    container = client.wait_success(container)
+
+    assert_fields(container, {
+        "type": "container",
+        "state": "stopped",
+        "imageUuid": uuid,
+        "firstRunning": None,
+    })
+
+    uuid = "localhost.localdomain:5000/sim:{}".format(random_num())
+    container = client.create_container(imageUuid=uuid,
+                                        startOnCreate=False)
+
+    container = client.wait_success(container)
+
+    assert_fields(container, {
+        "type": "container",
+        "state": "stopped",
+        "imageUuid": uuid,
+        "firstRunning": None,
+    })
+
+    # case 2: whitelist is set to different values, default set to null
+    whitelist_value = 'localhost.localdomain:5000'
+    whitelist_registries = admin_user_client.update(whitelist_registries,
+                                                    value=whitelist_value)
+    default_registry = admin_user_client.update(default_registry, value='')
+    uuid = "sim:{}".format(random_num())
+    # will fail, since default.registry is not set, by default it
+    # goes to dockerhub, which is not whitelisted
+    try:
+        client.create_container(imageUuid=uuid,
+                                startOnCreate=False)
+    except cattle.ApiError as e:
+        assert e.error.code == 'InvalidOption'
+
+    uuid = "namespace/sim:{}".format(random_num())
+    # will fail, since default.registry is not set, by default it
+    # goes to dockerhub, which is not whitelisted
+    try:
```

Comment:
Actually it is not a NIT. As we expect this failure for the test to be passed, you should put pytest.raises(ApiError) as e: